arch/x86/kernel/slmodule.c

```diff
@@ -147,104 +140,106 @@ static const struct file_operations sl_evtlog_ops = {
 	.llseek	= default_llseek,
 };
 
-static struct dentry *fs_entries[SL_FS_ENTRIES];
-
 struct sfs_file {
-	int parent;
 	const char *name;
 	const struct file_operations *fops;
 };
 
-static const struct sfs_file sl_files[] = {
-	{ SL_TXT_DIR_ENTRY, "sts", &sts_ops },
-	{ SL_TXT_DIR_ENTRY, "ests", &ests_ops },
-	{ SL_TXT_DIR_ENTRY, "errorcode", &errorcode_ops },
-	{ SL_TXT_DIR_ENTRY, "didvid", &didvid_ops },
-	{ SL_TXT_DIR_ENTRY, "ver_emif", &ver_emif_ops },
-	{ SL_TXT_DIR_ENTRY, "scratchpad", &scratchpad_ops },
-	{ SL_TXT_DIR_ENTRY, "e2sts", &e2sts_ops }
+#define SL_TXT_ENTRY_COUNT	7
+static const struct sfs_file sl_txt_files[] = {
+	{ "sts", &sts_ops },
+	{ "ests", &ests_ops },
+	{ "errorcode", &errorcode_ops },
+	{ "didvid", &didvid_ops },
+	{ "ver_emif", &ver_emif_ops },
+	{ "scratchpad", &scratchpad_ops },
+	{ "e2sts", &e2sts_ops }
 };
 
-static int sl_create_file(int entry, int parent, const char *name,
-			  const struct file_operations *ops)
-{
-	if (entry < 0 || entry > SL_TXT_DIR_ENTRY)
-		return -EINVAL;
-	fs_entries[entry] =
-		 securityfs_create_file(name, 0440,
-					fs_entries[parent], NULL, ops);
-	if (IS_ERR(fs_entries[entry])) {
-		pr_err("Error creating securityfs %s file\n", name);
-		return PTR_ERR(fs_entries[entry]);
-	}
-	return 0;
-}
+/* sysfs file handles */
+static struct dentry *slaunch_dir;
+static struct dentry *event_file;
+static struct dentry *txt_dir;
+static struct dentry *txt_entries[SL_TXT_ENTRY_COUNT];
 
 static long slaunch_expose_securityfs(void)
 {
 	long ret = 0;
-	int i = 0;
+	int i;
 
-	fs_entries[SL_ROOT_DIR_ENTRY] = securityfs_create_dir("slaunch", NULL);
-	if (IS_ERR(fs_entries[SL_ROOT_DIR_ENTRY])) {
-		pr_err("Error creating securityfs slaunch root directory\n");
-		ret = PTR_ERR(fs_entries[SL_ROOT_DIR_ENTRY]);
-		goto err;
-	}
+	slaunch_dir = securityfs_create_dir("slaunch", NULL);
+	if (IS_ERR(slaunch_dir))
+		return PTR_ERR(slaunch_dir);
 
 	if (slaunch_get_flags() & SL_FLAG_ARCH_TXT) {
-		fs_entries[SL_TXT_DIR_ENTRY] =
-			securityfs_create_dir("txt",
-					      fs_entries[SL_ROOT_DIR_ENTRY]);
-		if (IS_ERR(fs_entries[SL_TXT_DIR_ENTRY])) {
-			pr_err("Error creating securityfs txt directory\n");
-			ret = PTR_ERR(fs_entries[SL_TXT_DIR_ENTRY]);
-			goto err_dir;
+		txt_dir = securityfs_create_dir("txt", slaunch_dir);
+		if (IS_ERR(txt_dir)) {
+			ret = PTR_ERR(txt_dir);
+			goto remove_slaunch;
 		}
 
-		for (i = 0; i < SL_TXT_ENTRY_COUNT; i++) {
-			ret = sl_create_file(SL_TXT_FILE_FIRST - i,
-					     sl_files[i].parent, sl_files[i].name,
-					     sl_files[i].fops);
-			if (ret)
-				goto err_dir;
+		for (i = 0; i < ARRAY_SIZE(sl_files); i++) {
```

Comment:
I think you renamed this to sl_txt_files so this probably will fail to compile unless I am missing something.

Reply:
Not sure how that got passed since I compiled before push. Either way, will clean up.